pkg/build/builder/common.go

```diff
@@ -82,3 +91,141 @@ func updateBuildRevision(c client.BuildInterface, build *api.Build, sourceInfo *
 		glog.Warningf("An error occurred saving build revision: %v", err)
 	}
 }
+
+// randomBuildTag generates a random tag used for building images in such a way
+// that the built image can be referred to unambiguously even in the face of
+// concurrent builds with the same name in the same namespace.
+func randomBuildTag(namespace, name string) string {
+	repo := fmt.Sprintf("%s/%s", namespace, name)
+	randomTag := fmt.Sprintf("%08x", rand.Uint32())
+	maxRepoLen := reference.NameTotalLengthMax - len(randomTag) - 1
+	if len(repo) > maxRepoLen {
+		repo = fmt.Sprintf("%x", sha1.Sum([]byte(repo)))
+	}
+	return fmt.Sprintf("%s:%s", repo, randomTag)
+}
+
+// containerNamePrefix prefixes the name of containers launched by a build. We
+// cannot reuse the prefix "k8s" because we don't want the containers to be
+// managed by a kubelet.
+const containerNamePrefix = "openshift"
+
+// containerName creates names for Docker containers launched by a build. It is
+// meant to resemble Kubernetes' pkg/kubelet/dockertools.BuildDockerName.
```

Comment:
If for any reason a container with the same name exists, this will the error message printed in the build logs:

```
...
 ---> 44cebbfa95a0
Successfully built 44cebbfa95a0
I0205 15:33:44.079540       1 common.go:136] Running post commit hook with image demo/busybox-4:40c8782a ...
F0205 15:33:44.097297       1 builder.go:202] Error: build error: create container "openshift_docker-build_busybox-4_demo_post-commit": container already exists
```